arelle/ValidateXbrlCalcs.py

```diff
@@ -117,54 +145,86 @@ def validate(self):
+                                                boundIntervals[itemBindKey] = (x1 + min(y1,y2), x2 + max(y1,y2))
+                                                boundIntervalItems[itemBindKey].extend(_itemFacts)
                             for sumBindKey in boundSumKeys:
                                 ancestor, contextHash, unit = sumBindKey
                                 factKey = (sumConcept, ancestor, contextHash, unit)
                                 if factKey in self.sumFacts:
                                     sumFacts = self.sumFacts[factKey]
                                     for fact in sumFacts:
-                                        if fact in self.duplicatedFacts:
-                                            dupBindingKeys.add(sumBindKey)
-                                        elif sumBindKey not in dupBindingKeys and fact not in self.consistentDupFacts:
-                                            roundedSum = roundFact(fact, self.inferDecimals)
-                                            roundedItemsSum = roundFact(fact, self.inferDecimals, vDecimal=boundSums[sumBindKey])
-                                            if roundedItemsSum  != roundFact(fact, self.inferDecimals):
-                                                d = inferredDecimals(fact)
-                                                if isnan(d) or isinf(d): d = 4
-                                                _boundSummationItems = boundSummationItems[sumBindKey]
-                                                unreportedContribingItemQnames = [] # list the missing/unreported contributors in relationship order
-                                                for modelRel in modelRels:
-                                                    itemConcept = modelRel.toModelObject
-                                                    if (itemConcept is not None and 
-                                                        (itemConcept, ancestor, contextHash, unit) not in self.itemFacts):
-                                                        unreportedContribingItemQnames.append(str(itemConcept.qname))
-                                                self.modelXbrl.log('INCONSISTENCY', "xbrl.5.2.5.2:calcInconsistency",
-                                                    _("Calculation inconsistent from %(concept)s in link role %(linkrole)s reported sum %(reportedSum)s computed sum %(computedSum)s context %(contextID)s unit %(unitID)s unreportedContributingItems %(unreportedContributors)s"),
-                                                    modelObject=wrappedSummationAndItems(fact, roundedSum, _boundSummationItems),
+                                        if not fact.isNil:
+                                            if fact in self.duplicatedFacts:
+                                                dupBindingKeys.add(sumBindKey)
+                                            elif (xbrl21 and sumBindKey in boundSums and sumBindKey not in dupBindingKeys 
```

Comment:
This is fine, but it seems like it would be clearer to hoist the `if xbrl21` to the `for fact in sumFacts` so that `if xbrl21` and `if calc11` are at the same level.  (I think that's equivalent since `dupBindingKeys` is only used by this xbrl21 loop anyway?)

Reply:
The summation facts if xbrl21 and calc11 are at different levels because for xbrl21 the sum facts are processed one by one (with different effects when dups are allowed) whereas with calc11 the sum facts are processed as an intersection of consistent intervals at the top level.

Reply:
I meant that [this whole loop](https://github.com/Arelle/Arelle/blob/af5f948a53baac66132754e254fad24e7aeeda76/arelle/ValidateXbrlCalcs.py#L184) seems to do nothing unless `xbrl21` is true, so it would be clearer to hoist the `xbrl21` so that the two checks are at the same level:
```
                                if factKey in self.sumFacts:
                                    sumFacts = self.sumFacts[factKey]
                                    if xbrl21:
                                        for fact in sumFacts:
                                            ...
                                    if calc11:
                                        ...
```

Reply:
Ah, gotcha, done.